### PR TITLE
49108 Use existing create model for edit RNW

### PR DESCRIPTION
**Pull request by Ashish15251**

### CODE REVIEW

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/RNWRepository.cs [hunk 1]
```diff
@@ -83,23 +83,13 @@ public async Task<DateTime> GetRadioNavigationalWarningsLastModifiedDateTime()
             return DateTime.MinValue;
         }
 
-        public EditRadioNavigationalWarningAdmin GetRadioNavigationalWarningById(int id)
+        public RadioNavigationalWarning GetRadioNavigationalWarningById(int id)
         {
             RadioNavigationalWarning rnwWarning = _context.Set<RadioNavigationalWarning>().Find(id);
```
Comment (usha13529) [inline UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/RNWRepository.cs:88]: Can directly return result and then rnwWarning  variable can be removed.
```suggestion
            return _context.Set<RadioNavigationalWarning>().Find(id);
```

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/RNWRepository.cs [hunk 2]
```diff
@@ -83,23 +83,13 @@ public async Task<DateTime> GetRadioNavigationalWarningsLastModifiedDateTime()
...
-            rnw.IsDeleted = rnwWarning.IsDeleted;
-            return rnw;
+            return rnwWarning;
         }
 
-        public async Task UpdateRadioNavigationalWarning(EditRadioNavigationalWarningAdmin radioNavigationalWarningAdmin)
+        public async Task UpdateRadioNavigationalWarning(RadioNavigationalWarning radioNavigationalWarningAdmin)
         {
             RadioNavigationalWarning rnw = await _context.RadioNavigationalWarnings.FirstOrDefaultAsync(r => r.Id == radioNavigationalWarningAdmin.Id);
             rnw.WarningType = radioNavigationalWarningAdmin.WarningType;
```
Comment (usha13529) [inline UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/RNWRepository.cs:95]: No need to explicitly map values again. Please directly call update on DTO. Replace this block as mentioned below:

public async Task UpdateRadioNavigationalWarning(RadioNavigationalWarning radioNavigationalWarning)
        {
            radioNavigationalWarning.LastModified = DateTime.UtcNow;            
            _context.Update(radioNavigationalWarning);
            await _context.SaveChangesAsync();
        }

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/RNWRepository.cs [hunk 3]
```diff
@@ -83,23 +83,13 @@ public async Task<DateTime> GetRadioNavigationalWarningsLastModifiedDateTime()
...
-            rnw.Summary = rnwWarning.Summary;
-            rnw.Content = rnwWarning.Content;
-            rnw.ExpiryDate = rnwWarning.ExpiryDate;
-            rnw.IsDeleted = rnwWarning.IsDeleted;
-            return rnw;
+            return rnwWarning;
         }
 
-        public async Task UpdateRadioNavigationalWarning(EditRadioNavigationalWarningAdmin radioNavigationalWarningAdmin)
+        public async Task UpdateRadioNavigationalWarning(RadioNavigationalWarning radioNavigationalWarningAdmin)
```
Comment (usha13529) [inline UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/RNWRepository.cs:92]:
```suggestion
        public async Task UpdateRadioNavigationalWarning(RadioNavigationalWarning radioNavigationalWarning)
```

**Review by usha13529 — CHANGES_REQUESTED** [review]
Please fix the review comments.

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.IntegrationTests/RadioNavigationalWarnings/Admin/RadioNavigationalWarningEditAdminTest.cs [hunk 4]
```diff
@@ -68,73 +67,58 @@ public async Task WhenCallUpdateRadioNavigationWarnings_ThenRecordIsUpdated()
...
+        //        Id = 9,
+        //        WarningType = 1,
+        //        Reference = "ReferenceTest",
+        //        DateTimeGroup = new DateTime(2022, 12, 30),
+        //        Summary = "SummaryTest",
+        //        Content = "ContentTest",
+        //        ExpiryDate = new DateTime(2099, 1, 1),
+        //        IsDeleted = true
+        //    };
+        //}
```
[the remark this answers is not in the document]
Comment (Ashish15251) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.IntegrationTests/RadioNavigationalWarnings/Admin/RadioNavigationalWarningEditAdminTest.cs]: Changes done

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/RNWRepository.cs (hunk 1 above)
Comment (Ashish15251) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/RNWRepository.cs]: changes done

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/RNWRepository.cs (hunk 2 above)
Comment (Ashish15251) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/RNWRepository.cs]: changes done

**Review by usha13529 — APPROVED** [review]
Thanks for the changes @Ashish15251. Looks good to me.

**Review by usha13529 — APPROVED** [review]

**Review by barrie-cooper — APPROVED** [review]
LGTM, thanks @Ashish15251

**Pull request merged**